dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResultHydrated.java

```diff
@@ -0,0 +1,183 @@
+package com.dotcms.rendering.velocity.viewtools.navigation;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.velocity.tools.view.context.ViewContext;
+
+import com.liferay.portal.model.User;
+
+public final class NavResultHydrated extends NavResult{
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    final NavResult navResult;
+    final transient ViewContext context;
+
+    public NavResultHydrated(final NavResult navResult, final ViewContext context) {
+        super(navResult);
+        this.navResult = navResult;
+        this.context = context;
+
+    }
+
+
+
+    public boolean isActive() {
+        if (context != null && UtilMethods.isSet(navResult.getHref())) {
+            HttpServletRequest req = (HttpServletRequest) context.getRequest();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "HttpServletRequest". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)